modules/host/announce.go

```diff
@@ -67,3 +59,28 @@ func (h *Host) Announce() error {
 
 	return nil
 }
+
+// Announce creates a host announcement transaction, adding information to the
+// arbitrary data, signing the transaction, and submitting it to the
+// transaction pool.
+func (h *Host) Announce() error {
+	// check that our address is reachable
+	lockID := h.mu.RLock()
+	addr := h.myAddr
+	h.mu.RUnlock(lockID)
+	if addr.Host() == "" {
+		return errors.New("can't announce without knowing external IP")
```

Comment:
I am not sure.

On one hand, --force means "do it I know what I'm doing". On the other, it's pretty hard to imagine that you are doing the right thing if your addr is `""`